src/api/user_presence.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::clock::ClockInt;
+use crate::ctap::Channel;
```

Comment:
I don't think `api` (and `env`) should depend on `ctap`. It should ideally be the opposite. Should we move those definitions (`Channel` and `ChannelID`) to `api`? In which case, I would call the `send_or_recv` trait `transport` or `usb_hid`.

Reply:
Hmm, I was looking at customization.rs and concluded that it's ok to `use crate::ctap::...` in `api`.
If that's not advisable, then it's certainly possible to move `Channel` and `ChannelID` to `api` module.
If I understand correctly, new variants like `NFC` and `BLE` are supposed to be added to `Channel` enum in the future. In this case, it would be logical to rename `ChannelID` to something like `HidChannelID`, as that type is HID-specific.

Reply:
It is technically already a `hid::ChannelID`, but we import it in a way that drops `hid`. If we move it, renaming is good. But since there is precedence, that's more of a beautification we can do now or later.

However, this made me think of a bigger naming concern I have (that is easy to fix as I see it). When I designed `Transport` and `Channel`, I intended `Transport` to be the external concept of "a physical connection" and `Channel` should be "a logical connection to one client". `Transport` is an external notion, `Channel` only known internally to the library. Users don't have to understand how we use channel IDs and put them into packets. So your API should never use `Channel`, but always `Transport`. The one `Channel` I found that is left is actually unused in `TockEnv` and I think can be removed. The `TestEnv` usage seems to be trivial. The button check function does not need to know what channel it is checking for.

On top of that, there are 2 things that are called channel now: My `Channel`, and the `CtapHidChannel` that is something else entirely.
- It is something per `Transport`, not per `Channel`.
- It's not a pure data container like `Channel` and `Transport`.
May I therefore suggest that we rename it to "HidConnection" or any other similar word you like that is not "Channel"?

I know I notice that a bit late, but I noticed this is a bit confusing as is. So the 2 action items:
- Remove `Channel` from this API, it's not needed.
- Rename `CtapHidChannel`.

Reply:
> I was looking at customization.rs and concluded that it's ok to use crate::ctap::... in api

Yes, we should ideally move those enums out of ctap into api.

Reply:
That's for a future PR then! The `Channel` dependency is gone.